zeppelin-server/src/main/java/org/apache/zeppelin/realm/LdapRealm.java

```diff
@@ -128,6 +128,8 @@
   private static final String SUBJECT_USER_GROUPS = "subject.userGroups";
   private static final String MEMBER_URL = "memberUrl";
   private static final String POSIX_GROUP = "posixGroup";
+  private static final String MATCHING_RULE_IN_CHAIN_FORMAT = 
+      "(&(objectClass=%s)(%s:1.2.840.113556.1.4.1941:=%s))";
```

Comment:
A comment about this magic number would be helpful in future debug. Also if you can specify the doc URL would be helpful.
I understand its `This rule is limited to filters that apply to the DN. This is a special "extended" match operator that walks the chain of ancestry in objects all the way to the root until it finds a match.`